swiftsimio/subset_writer.py

```diff
@@ -0,0 +1,159 @@
+"""
+Contains functions for reading a subset of a SWIFT dataset and writing
+it to a new file.
+"""
+
+from swiftsimio.reader import SWIFTUnits, SWIFTMetadata
+from swiftsimio.masks import SWIFTMask
+from swiftsimio.accelerated import read_ranges_from_file
+import swiftsimio.metadata as metadata
+
+import unyt
+import h5py
+import numpy as np
+
+
+def get_dataset_mask(mask: SWIFTMask, dataset_name: str, suffix=""):
+    """
+    Return appropriate mask or mask size for given dataset
+
+    Parameters
+    ----------
+    mask : SWIFTMask
+        the mask used to define subset that is written to new snapshot
+    dataset_name : str
+        the name of the dataset we're interested in. This is the name from the
+        hdf5 file (i.e. "PartType0", rather than "gas")
+    suffix : str, optional
+        specify a suffix string to append to dataset underscore name to return
+        something other than the dataset mask. This is specifically used for
+        returning the mask size by setting suffix="_size", which would return,
+        for example mask.gas_size
+
+    Returns
+    -------
+    np.ndarray
+        mask for the appropriate dataset
+
+    """
+    if "Cells" in dataset_name:
+        return None
+    elif "PartType" in dataset_name:
+        part_type = [int(x) for x in filter(str.isdigit, dataset_name)][0]
+        mask_name = metadata.particle_types.particle_name_underscores[part_type]
+        return (
+            getattr(mask, f"{mask_name}{suffix}") if hasattr(mask, mask_name) else None
```

Comment:
You can provide None as the third attribute to `getattr` and it will return that if it does not find the attribute. You are also not checking for a consistent thing here.

Reply:
Done.